clients/hadoopfs/src/test/java/io/lakefs/committer/LakeFSOutputCommitterTest.java

```diff
@@ -0,0 +1,155 @@
+    protected static final String S3_ACCESS_KEY_ID = "AKIArootkey";
+    protected static final String S3_SECRET_ACCESS_KEY = "secret/minio/key=";
+
+    @Rule
+    public final GenericContainer s3 = new GenericContainer(MINIO.toString()).
+        withCommand("minio", "server", "/data").
+        withEnv("MINIO_ROOT_USER", S3_ACCESS_KEY_ID).
+        withEnv("MINIO_ROOT_PASSWORD", S3_SECRET_ACCESS_KEY).
+        withEnv("MINIO_DOMAIN", "s3.local.lakefs.io").
+        withEnv("MINIO_UPDATE", "off").
+        withExposedPorts(9000);
+
+
+    @After
+    public void closeSparkSession(){
+        spark.close();
+    }
+
+
+    @Before
+    public void setUp() throws Exception {
+        AWSCredentials creds = new BasicAWSCredentials(S3_ACCESS_KEY_ID, S3_SECRET_ACCESS_KEY);
+
+        ClientConfiguration clientConfiguration = new ClientConfiguration()
+                .withSignerOverride("AWSS3V4SignerType");
+        String s3Endpoint = String.format("http://s3.local.lakefs.io:%d", s3.getMappedPort(9000));
+
+        s3Client = new AmazonS3Client(creds, clientConfiguration);
+
+        S3ClientOptions s3ClientOptions = new S3ClientOptions()
+            .withPathStyleAccess(true);
+        s3Client.setS3ClientOptions(s3ClientOptions);
+        s3Client.setEndpoint(s3Endpoint);
+
+        s3Bucket = "example";
+        s3Base = String.format("s3://%s", s3Bucket);
+        CreateBucketRequest cbr = new CreateBucketRequest(s3Bucket);
+        s3Client.createBucket(cbr);
+
+        SparkConf sparkConf = new SparkConf().setMaster("local").setAppName("Spark test")
+                .set("spark.sql.shuffle.partitions", "17")
+                .set("spark.hadoop.mapred.output.committer.class","io.lakefs.committer.DummyOutputCommitter")
+                .set("spark.hadoop.fs.lakefs.impl","io.lakefs.LakeFSFileSystem")
+                .set("fs.lakefs.access.key", "AKIAIOSFODNN7EXAMPLE")
+                .set("fs.lakefs.secret.key", "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY")
+                .set("fs.lakefs.endpoint", "http://localhost:8000/api/v1")
+                .set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+                .set(org.apache.hadoop.fs.s3a.Constants.ACCESS_KEY, S3_ACCESS_KEY_ID)
+                .set(org.apache.hadoop.fs.s3a.Constants.SECRET_KEY, S3_SECRET_ACCESS_KEY)
+                .set(org.apache.hadoop.fs.s3a.Constants.ENDPOINT, s3Endpoint);
+        this.spark = SparkSession.builder().config(sparkConf).getOrCreate();
+    }
+
+
+    @Test
+    public void testWriteTextfile() throws ApiException, IOException {
+        String output = "lakefs://example/main/some_df";
+
+        StructType structType = new StructType();
+        structType = structType.add("A", DataTypes.StringType, false);
+
+        List<Row> nums = new ArrayList<Row>();
+        nums.add(RowFactory.create("value1"));
+        nums.add(RowFactory.create("value2"));
```

Comment:
Again, I probably prefer
```suggestion
        List<Row> nums = ImmutableList.of(
        RowFactory.create("value1"),
        RowFactory.create("value2"));
```
using ImmutableList from Guava (Google Commons).